src/Streamly/Streams/StreamD.hs

```diff
@@ -4237,3 +4241,22 @@ foldParallel f m = do
     sv <- newParallelVar StopNone defState
     toSVarParallel defState sv (toStreamD m)
     f $ fromStreamD $ fromSVar sv
+
+-- XXX This is taking too much time
+-- XXX This seems to fuse as far as I've checked
+-- | Take last 'n' elements from the stream and discard the rest.
+{-# INLINABLE lastN #-}
+lastN :: (Storable a, MonadIO m) => Int -> Fold m a (Array a)
```

Comment:
1) It will be nice to have a test for this combinator.
2) Please use 4 space indentation consistent with the rest of the library.